spec/std/openssl/ssl/hostname_validation_spec.cr

```diff
@@ -0,0 +1,54 @@
+require "spec"
+require "openssl/ssl/hostname_validation"
+
+describe OpenSSL::SSL::HostnameValidation do
+  describe "matches_hostname?" do
+    it "matches ipv4, ipv6" do
+      OpenSSL::SSL::HostnameValidation.matches_hostname?("192.168.1.1", "192.168.1.1").should be_true
+      OpenSSL::SSL::HostnameValidation.matches_hostname?("192.168.1.2", "192.168.1.1").should be_false
+      OpenSSL::SSL::HostnameValidation.matches_hostname?("::1", "::1").should be_true
+      OpenSSL::SSL::HostnameValidation.matches_hostname?("::1", "::2").should be_false
```

Comment:
The one I mentioned plus `fe80::1`, `fe80::1:2`, `::1:2`, `fe80::`, `::` and their expanded versions are probably enough.